### PR TITLE
CSB-29 Replace Dataframes with Datasets

### PR DESCRIPTION
**Pull request by jan-Kulpas**

### CODE REVIEW

File: model_utilities/config.ini [hunk 1]
```diff
@@ -9,8 +9,10 @@ output_model_name = pl-csb-model-v1
 batch_size = 8
 max_length = 128
 warmup_steps = 100
-training_steps = 1000
+num_epochs = 3
```
Comment (filip-szweda) [inline model_utilities/config.ini:12]: Are 3 epochs comparable to 1000 steps?

File: model_utilities/train/finetuner.py [hunk 2]
```diff
@@ -56,45 +41,58 @@ def __plot_losses(self, losses: list[float]) -> None:
...
-            except Exception as e:
-                optimizer.zero_grad(set_to_none=True)
-                x, y, loss = None, None, None
-                self.__cleanup()
-                self.__logger.error("Error: unexpected exception during training, exception: %s", str(e))
-                continue
+        num_epochs = int(train_conf["num_epochs"])
+        num_training_steps = len(batched_dataset)
+
+        self.__logger.critical(num_training_steps)
```
Comment (filip-szweda) [inline model_utilities/train/finetuner.py:62]: > CRITICAL | A serious error, indicating that the program itself may be unable to continue running.

https://docs.python.org/3/howto/logging.html
Use debug or info level instead.

File: model_utilities/train/finetuner.py [hunk 3]
```diff
@@ -56,45 +41,58 @@ def __plot_losses(self, losses: list[float]) -> None:
...
+
+        progress_bar = trange(num_epochs * num_training_steps)
+        for epoch in range(num_epochs):
+            for i in range(num_training_steps):
```
Comment (filip-szweda) [inline model_utilities/train/finetuner.py:66]:
```suggestion
            for _ in range(num_training_steps):
```

File: model_utilities/config.ini (hunk 1 above)
Comment (jan-Kulpas) [reply model_utilities/config.ini:12]: Depends on the size of the dataset, with the current one it's technically 18x more, nevertheless 1000 steps is not a good training length anyway and was chosen for unknown reason god knows when.

File: model_utilities/train/finetuner.py (hunk 2 above)
Comment (jan-Kulpas) [reply model_utilities/train/finetuner.py:62]: Forgot to change back.

File: model_utilities/train/finetuner.py (hunk 3 above)
Comment (jan-Kulpas) [reply model_utilities/train/finetuner.py:66]: `i` is used for batch index

**Review by filip-szweda — APPROVED** [review]

**Pull request merged**